dbms/src/DataStreams/HashJoinProbeBlockInputStream.cpp

```diff
@@ -67,17 +67,34 @@ Block HashJoinProbeBlockInputStream::getHeader() const
 
 Block HashJoinProbeBlockInputStream::readImpl()
 {
-    if (probe_process_info.all_rows_joined_finish)
+    // if join finished, return {} directly.
+    if (squashing_transform.isJoinFinished())
+    {
+        return Block{};
+    }
+
+    while (squashing_transform.needAppendBlock())
+    {
+        Block result_block = getOutputBlock(probe_process_info);
+        squashing_transform.appendBlock(result_block);
+    }
+    return squashing_transform.getFinalOutputBlock();
+}
+
+Block HashJoinProbeBlockInputStream::getOutputBlock(ProbeProcessInfo & probe_process_info_) const
```

Comment:
Here it is not necessary to pass `probe_process_info_`
```suggestion
Block HashJoinProbeBlockInputStream::getOutputBlock() const
```

Reply:
done.